### DISCUSSION TITLE
`docker manifest inspect` currently failing on some GHCR images

### DISCUSSION THREAD

**Discussion by pedrolamas** [community/community#45779] [Actions]
### Select Topic Area

Bug

### Body

Until yesterday, I was using `docker manifest inspect ghcr.io/USER/CONTAINER:TAG` to check if a specific container tag already existing, but that now seems to have stopped working...

Specifically, my [docker-klipper-simulavr](https://github.com/pedrolamas/docker-klipper-simulavr/pkgs/container/docker-klipper-simulavr) image doesn't even show for "latest" tag:

```sh
> docker manifest inspect ghcr.io/pedrolamas/docker-klipper-simulavr:latest
no such manifest: ghcr.io/pedrolamas/docker-klipper-simulavr:latest
```

The above now just returns "no such manifest: ghcr.io/pedrolamas/docker-klipper-simulavr:latest" when the image is actually there and properly tagged with "latest"!

`docker pull ghcr.io/pedrolamas/docker-klipper-simulavr:latest` works fine and I can even access it directly via browser, but sometimes I do get a "tag does not exist" red warning from GitHub when I do that...

Any idea of what is going on or where should I report issues with GHCR ?

**Comment by ChristopherHX** [comment]
_No idea what is going on, however there is an alternative docker tool to do what you want_

You can use `docker buildx imagetools inspect ghcr.io/pedrolamas/docker-klipper-simulavr:latest` it just works fine.

I can confirm that `docker manifest` doesn't find your image on my machine.

_I prefer `docker buildx imagetools` for publishing multiarch images from a local docker registry over `docker manifest` (it refuses to copy images from the source to the target registry)._

**Reply by pedrolamas** [reply]
Thank you @ChristopherHX: as a way to mitigate the problem, indeed that is a working solution!

https://github.com/pedrolamas/docker-klipper-simulavr/actions/runs/4043447885/jobs/6952373086#step:3:5

But unless someone points me somewhere saying that `docker manifest inspect` is now deprecated, I do feel that it should also work (as it was working fine just a couple of days ago) so I will keep this open for now so that other people can also see and reply.

**Reply by ChristopherHX** [reply]
It could also be that `docker manifest` is still EXPERIMENTAL and fails to parse your image manifest.

`docker manifest` still works for my own docker images
```
~$ docker manifest inspect ghcr.io/christopherhx/ubuntu:act-latest-dev
{
   "schemaVersion": 2,
   "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
   "manifests": [
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 2004,
         "digest": "sha256:c01c979206796d5dc96f3a43786544a922aa338ac64ecf3565620f7d25ef53ba",
         "platform": {
            "architecture": "amd64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 2004,
         "digest": "sha256:c00c7ad83a22324da2905d9cc51176174f80ec44234274af553deff0fbdd3ad8",
         "platform": {
            "architecture": "arm64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 2004,
         "digest": "sha256:6ec7092cbe8bb86b8ffe7899f32f94c2a37c868a7c061019bb442cde1d3e6fe3",
         "platform": {
            "architecture": "arm",
            "os": "linux",
            "variant": "v7"
         }
      }
   ]
}
```

I noticed your image manifest has a special submanifest, which my manifests doesn't use:
```
~$ docker buildx imagetools inspect ghcr.io/pedrolamas/docker-klipper-simulavr:latest
Name:      ghcr.io/pedrolamas/docker-klipper-simulavr:latest
MediaType: application/vnd.oci.image.index.v1+json
Digest:    sha256:c6dec743f16ab2ef71fbbb3310389ea4396fad4427363b4e2c084405f916fef4

Manifests:
  Name:      ghcr.io/pedrolamas/docker-klipper-simulavr:latest@sha256:70320fdcd5ea1f9bf12eba64fa8e8e2672b47effb8df59d4c65ef291138ab981
  MediaType: application/vnd.oci.image.manifest.v1+json
  Platform:  linux/amd64

  Name:      ghcr.io/pedrolamas/docker-klipper-simulavr:latest@sha256:40f6987c8dfe35c5692dd1f419681180bdd2076759d5bebbb6266ced8d958ca5
  MediaType: application/vnd.oci.image.manifest.v1+json
  Platform:  unknown/unknown
    vnd.docker.reference.digest: sha256:70320fdcd5ea1f9bf12eba64fa8e8e2672b47effb8df59d4c65ef291138ab981
    vnd.docker.reference.type:   attestation-manifest
````
see second child submanifest.

**Reply by pedrolamas** [reply]
Indeed, I do see the "attestation-manifest" and I have no idea what it is, but that does seem to be "a thing", considering it also shows on the [builx imagetools inspect docs](https://docs.docker.com/engine/reference/commandline/buildx_imagetools_inspect/)

Comparing your build workflow with mine, the only major difference is that I am just using Docker GitHub Actions (like `docker/build-push-action@v3`) instead of docker commands...

Makes me wonder if there has been an update on these actions that are now causing issues...

**Reply by pedrolamas** [reply]
I've found what that extra manifest it: https://docs.docker.com/build/attestations/attestation-storage/

**Reply by pedrolamas** [reply]
I have a feeling I have found the source of the problem: 2 weeks ago there the `build-push-action` action released [v3.3.0](https://github.com/docker/build-push-action/releases/tag/v3.3.0) that seems to have added a new "attestation" that is not fully compatible with GHCR!

I've found [this report](https://github.com/docker/build-push-action/issues/771) or issues with this change, which I think it might be related to the whole situation.

**Reply by djmaze** [reply]
Ah, so people should set `provenance: false` when building their images on Github [as shown here](https://github.com/docker/build-push-action/issues/771#issuecomment-1400925570).

**Reply by mehulrajdev** [reply]
https://github.com/mehulrajdev/Zayn-Malik

**Comment by mehulrajdev** [comment]
![Visitor Count](https://profile-counter.glitch.me/{YOUR USER}/count.svg)
